PukingPredator/Assets/Scripts/Timer.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Timer : MonoBehaviour
+{
+    public float duration = 0f;
+    private bool isRunning = false;
+    private float timeRemaining;
+
+    public EventHandler OnTimerComplete;
+
+    public void Init(float timerDuration)
+    {
+        duration = timerDuration;
+        timeRemaining = duration;
+    }
+
+    void Update()
```

Comment:
would a coroutine with waitforseconds be more efficient than checking every frame for the timer?

Reply:
good point, added the coroutine

Reply:
When you swapped it to use a coroutine, you left in the property "timeRemaining" which is no longer actually the time remaining and it isn't currently used anywhere in the project. That property should've been removed if we were just swapping to a coroutine entirely.

In reality, the time remaining is necessary so we can show how close something is to decaying (whether we plan to use a counting down circle like in my sketches or a visible number or anything else), so this change should be undone.